src/main/api/ledger/evm/address.ts

```diff
@@ -0,0 +1,184 @@
+import type Transport from '@ledgerhq/hw-transport'
+import { defaultArbParams } from '@xchainjs/xchain-arbitrum'
+import { defaultBscParams } from '@xchainjs/xchain-bsc'
+import { Network } from '@xchainjs/xchain-client'
+import { defaultEthParams } from '@xchainjs/xchain-ethereum'
+import { ClientLedger, LedgerSigner } from '@xchainjs/xchain-evm'
+import { Chain } from '@xchainjs/xchain-util'
+import * as E from 'fp-ts/Either'
+
+import { LedgerError, LedgerErrorId } from '../../../../shared/api/types'
+import { defaultAvaxParams } from '../../../../shared/avax/const'
+import { getDerivationPath, getDerivationPaths } from '../../../../shared/evm/ledger'
+import { EvmHDMode } from '../../../../shared/evm/types'
+import { isError } from '../../../../shared/utils/guard'
+import { WalletAddress } from '../../../../shared/wallet/types'
+
+export const getEVMAddress = async ({
+  chain,
+  transport,
+  walletAccount,
+  walletIndex,
+  evmHDMode,
+  network
+}: {
+  chain: Chain
+  transport: Transport
+  walletAccount: number
+  walletIndex: number
+  evmHDMode: EvmHDMode
+  network: Network
+}): Promise<E.Either<LedgerError, WalletAddress>> => {
+  let clientParams
+
+  switch (chain) {
+    case 'ETH':
```

Comment:
What about `ETHChain`, `ARBChain`, etc instead of string?

Reply:
yep